### PR TITLE
Sem 46 remove circular references in sis data

### PR DESCRIPTION
**Pull request by mcginkel**
Removed circular dependency in objects and calls
Changed some locations where group is still used
Added name (first,prefix,sur)
Removed enrollment

This also contains the changes that were discussed 22-feb-2021

### CODE REVIEW

File: reference/sisdata.v1.yaml [hunk 1]
```diff
@@ -96,6 +96,12 @@ components:
           type: string
         eckid:
           type: string
+        firstname:
```
Comment (mcginkel) [inline reference/sisdata.v1.yaml:99]: Added these for use in LMS, LA and MP can still use displayname

File: reference/sisdata.v1.yaml [hunk 2]
```diff
@@ -105,14 +111,37 @@ components:
         groups:
           type: array
           items:
-            $ref: '#/components/schemas/group'
+            $ref: '#/components/schemas/groupReference'
```
Comment (mcginkel) [inline reference/sisdata.v1.yaml:114]: Included the reference to a group (and minimal data to create a group)

File: reference/sisdata.v1.yaml [hunk 3]
```diff
@@ -173,14 +230,22 @@ components:
...
-      title: teacherOfGroup
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+      required:
+        - id
+        - name
+    teacherOfClass:
```
Comment (mcginkel) [inline reference/sisdata.v1.yaml:247]: Teacher of group or class is not used anymore, can we remove this ?

Comment (cliftonc) [reply reference/sisdata.v1.yaml:251]: Fine by me.

**Review by cliftonc — APPROVED** [review]

**Review by edwinverwoerd — APPROVED** [review]

**Pull request merged**